engine/src/cmd/unit_json_factory.cpp

```diff
@@ -39,7 +38,12 @@
 void UnitJSONFactory::ParseJSON(VSFileSystem::VSFile &file, bool player_ship) {
     const std::string json_text = file.ReadFull();
 
-    boost::json::value json_value = boost::json::parse(json_text);
+    boost::json::value json_value;
+    try {
+        json_value = boost::json::parse(json_text);
+    } catch (std::exception const& e) {
```

Comment:
Let's try to be more consistent about putting try-catch blocks around code that could reasonably be expected to throw.